kymatio/scattering1d/scattering1d.py

```diff
@@ -233,36 +233,42 @@ def build(self):
         self.phi_f = phi_f
         self._type(torch.FloatTensor)
 
-    def _type(self, target_type):
-        """Change the datatype of the filters
-
-        This function is used internally to convert the filters. It does not
-        need to be called explicitly.
-
-        Parameters
-        ----------
-        target_type : type
-            The desired type of the filters, typically `torch.FloatTensor`
-            or `torch.cuda.FloatTensor`.
+    def _apply(self, fn):
         """
-        cast_psi(self.psi1_f, target_type)
-        cast_psi(self.psi2_f, target_type)
-        cast_phi(self.phi_f, target_type)
+            Mimicates the behavior of the function _apply() of a nn.Module()
+        """
+        _apply_psi(self.psi1_f, fn)
+        _apply_psi(self.psi2_f, fn)
+        _apply_phi(self.phi_f, fn)
         return self
 
-    def cpu(self):
-        """Move to the CPU
+    def cuda(self, device=None):
+        """
+            Mimicates the behavior of the function cuda() of a nn.Module()
+        """
+        return self._apply(lambda t: t.cuda(device))
 
-        This function prepares the object to accept input Tensors on the CPU.
+    def to(self, *args, **kwargs):
+        """
+            Mimicates the behavior of the function to() of a nn.Module()
         """
-        return self._type(torch.FloatTensor)
+        device, dtype, non_blocking = torch._C._nn._parse_to(*args, **kwargs)
```

Comment:
OK, so here we are using torch. I guess we have to talk about how to separate this away to have different backends/frontends?

Reply:
Yes, it will happen in some future commits!!